### PR TITLE
feat: renderer-to-image service

### PR DESCRIPTION
**Pull request by HJunyuan**
# Renderer-to-Image Service

An API endpoint for users to obtain an image screenshot of their Open Attestation document.

Examples

### Generating your query params

The query parms used in this API endpoint closely resembles the standards set out in this document: [Universal Actions for Open-Attestation Documents](https://github.com/Open-Attestation/adr/blob/master/universal_actions.md#universal-actions-for-open-attestation-documents)

With the exception of the `#` anchor which is now embedded as a query param called `anchor`. As such, this API endpoint expects 2 query params:

1. Decoded resource after `?q=`:

   ```json
   {
     "payload": {
       "uri": "https://url-to-document.example.com"
     }
   }
   ```

2. Decoded resource after `&anchor=`:

   ```json
   {
     "key": "secret-key"
   }
   ```

   \*The `key` object is added as a query param because anchors do not get sent to the server-side.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,47 +1,33 @@
...
+# Renderer-to-Image Service
 
-This is a really simple project that shows the usage of Next.js with TypeScript.
+An API endpoint for users to obtain a image screenshot of their Open Attestation document.
 
-## Preview
+## Getting Started
 
-Preview the example live on [StackBlitz](http://stackblitz.com/):
+Try it out here: `https://example.com/api/v2/renderImage?q={}&anchor={}` // TODO: Deploy on Netlify
```
Comment (Nebulis) [inline README.md]: to update

File: README.md [hunk 2]
```diff
@@ -1,47 +1,33 @@
...
 
-[![Open in StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/vercel/next.js/tree/canary/examples/with-typescript)
+### Generating your query params
 
-## Deploy your own
+The query parms used in this API endpoint closely resembles the standards set out in this document: [Universal Actions for Open-Attestation Documents](https://github.com/Open-Attestation/adr/blob/master/universal_actions.md#universal-actions-for-open-attestation-documents)
```
Comment (Nebulis) [inline README.md]: can direct to toolkit

Comment (Nebulis) [reply README.md]: or maybe not :D

File: package.json [hunk 3]
```diff
@@ -4,17 +4,40 @@
...
     "start": "next start",
-    "type-check": "tsc"
+    "lint": "next lint",
+    "type-check": "tsc",
+    "test": "jest"
   },
   "dependencies": {
+    "@govtechsg/decentralized-renderer-react-components": "^3.5.2",
+    "@govtechsg/oa-encryption": "^1.3.3",
+    "@govtechsg/open-attestation": "^5.4.0",
```
Comment (Nebulis) [inline package.json]: can use 6

File: pages/api/v1/renderImage.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import { NextApiRequest, NextApiResponse } from "next";
```
Comment (Nebulis) [inline pages/api/v1/renderImage.ts]: sorry to botherbut can use kebab-case for all files when possible ?

**Comment by HJunyuan** [comment]
@Nebulis Shall we merge the first version in? 🤓

**Comment by Nebulis** [comment]
Let's goooo

**Review by Nebulis — APPROVED** [review]

**Pull request merged**